DEVELOPMENT_GUIDE.md

```diff
@@ -116,30 +95,116 @@ This step is optional and will use the specified version of
+# Using SSH
+$ git clone git@github.com:aws/serverless-application-model.git
+# Using HTTPS
+$ git clone https://github.com/aws/serverless-application-model.git
+$ cd serverless-application-model
+$ git checkout develop
+```
+
+Make sure you are in the virtual environment. 
+Install the SAM Transformer in editable mode so that 
+all changes you make to the SAM Transformer locally are immediately picked up for SAM CLI. 
+
+```sh
+(samcli38) $ pip install -e .
+```
 
-``git clone https://github.com/awslabs/serverless-application-model/``
+Move back to your SAM CLI directory and re-run init, If necessary: open requirements/base.txt and replace the version number of aws-sam-translator with the ``version number`` specified in your local version of `serverless-application-model/samtranslator/__init__.py`
 
-``git checkout develop ``
+```sh
+# assuming your AWS SAM CLI repository is cloned into ~/projects/aws-sam-cli
+(samcli38) $ cd ~/projects/aws-sam-cli
+(samcli38) $ make init
+```
 
-Install the SAM Transformer in editable mode so that all changes you make to the SAM Transformer locally are immediately picked up for SAM CLI. 
+## Making a Pull Request
 
-``pip install -e . `` 
+Above demonstrates how to setup the environment, which is enough
+to play with the AWS SAM CLI source code. However, if you want to
+contribute to the repository, there are a few more things to consider.
 
-Move back to your SAM CLI directory and re-run init, If necessary: open requirements/base.txt and replace the version number of aws-sam-translator with the ``version number`` specified in your local version of `serverless-application-model/samtranslator/__init__.py`
+### Make Sure AWS SAM CLI Work in Multiple Python Versions
+
+We support 3.6, 3.7 and 3.8 versions. Our CI/CD pipeline is setup to run
+unit tests against all Python versions. So make sure you test it
+with all versions before sending a Pull Request.
+See [Unit testing with multiple Python versions](#unit-testing-with-multiple-python-versions).
 
-``cd ../aws-sam-cli``
- 
-``make init``
+If you chose to use `pyenv` in the previous session, setting up a 
+different Python version should be easy:
 
-Running Tests
--------------
+(assuming you are in virtual environment `samcli38`)
 
-### Unit testing with one Python version
+```sh
+(samcli38) $ pyenv deactivate samcli38
+$ pyenv install 3.7.10  # one time setup
+$ pyenv pyenv virtualenv 3.7.10 samcli37   # one time setup
+$ pyenv activate samcli37
+(samcli37) $ python --version
+(samcli37) $ make init  # one time setup
+Python 3.7.10
+```
+
+### Format Python Code
+
+We format our code using [Black](https://github.com/python/black) and verify the source code is
+black compliant in AppVeyor during PRs. Black will be installed automatically with `make init`.
+
+There are generally 3 options to make sure your change is compliant with our formatting standard:
+
+#### (Optional 1) Run `make black`
```

Comment:
Should it be `Option 1` or `Optional 1`?

Reply:
good catch, updated